### ISSUE TITLE
Only send single fully complete forms at a time

### ISSUE THREAD

**Issue by plexus** [plexus/unrepl.el#3]
At the moment each line is sent as is, I think that'll make it hard for us to properly track evaluation. Better would be to use `scan-sexps` to only read and send over a single s-exp at a time.

**Comment by bbatsov** [comment]
Yeah, that's basically what CIDER's REPL does. And another argument for trying reuse higher-level functionality instead of implementing it again and again in every similar package. Perhaps you can also use `comint` like recent versions of SLIME, but there are certain limitations there as well.
